src/__swaps__/screens/Swap/hooks/useSwapWarning.ts

```diff
@@ -38,175 +41,152 @@ export interface SwapTimeEstimate {
 
 type UsePriceImpactWarningProps = {
   SwapInputController: ReturnType<typeof useSwapInputsController>;
+  inputAsset: SharedValue<ExtendedAnimatedAssetWithColors | null>;
+  outputAsset: SharedValue<ExtendedAnimatedAssetWithColors | null>;
+  quote: SharedValue<Quote | CrosschainQuote | QuoteError | null>;
   sliderXPosition: SharedValue<number>;
   isFetching: SharedValue<boolean>;
 };
 
-export const useSwapWarning = ({ SwapInputController, isFetching, sliderXPosition }: UsePriceImpactWarningProps) => {
+type CurrentProps = {
+  inputNativeValue: string | number;
+  outputNativeValue: string | number;
+  quote: Quote | CrosschainQuote | QuoteError | null;
+  isFetching: boolean;
+  sliderXPosition: number;
+};
+
+export const useSwapWarning = ({
+  SwapInputController,
+  inputAsset,
+  outputAsset,
+  quote,
+  isFetching,
+  sliderXPosition,
+}: UsePriceImpactWarningProps) => {
   const { nativeCurrency: currentCurrency } = useAccountSettings();
+  const red = useForegroundColor('red');
+  const orange = useForegroundColor('orange');
+  const label = useForegroundColor('labelTertiary');
 
+  const swapWarning = useSharedValue<SwapWarning>({ type: SwapWarningType.none, title: '', color: label, icon: '', subtitle: '' });
+
+  // TODO: Can remove this if not needed elsewhere, but thought it might be useful.
   const timeEstimate = useSharedValue<SwapTimeEstimate | null>(null);
 
-  const swapWarning = useSharedValue<SwapWarning>({
-    type: SwapWarningType.none,
-    display: '',
-  });
+  const colorMap: Record<string, string> = useMemo(
+    () => ({
+      [SwapWarningType.severe]: red,
+      [SwapWarningType.unknown]: red,
+      [SwapWarningType.long_wait]: orange,
+      [SwapWarningType.none]: label,
+      [SwapWarningType.high]: orange,
+
+      // MARK: swap quote errors
+      [SwapWarningType.no_quote_available]: label,
+      [SwapWarningType.insufficient_liquidity]: label,
+      [SwapWarningType.fee_on_transfer]: label,
+      [SwapWarningType.no_route_found]: label,
+    }),
+    [label, orange, red]
+  );
 
-  const getWarning = useCallback(
-    ({
-      inputNativeValue,
-      outputNativeValue,
-      quote,
-      isFetching,
-    }: {
-      inputNativeValue: BigNumberish;
-      outputNativeValue: BigNumberish;
-      quote: Quote | CrosschainQuote | QuoteError | null;
-      isFetching: boolean;
-    }) => {
-      const updateWarning = (values: SwapWarning) => {
-        'worklet';
-        swapWarning.modify(prev => ({
-          ...prev,
-          ...values,
-        }));
-      };
+  const updateWarning = useCallback(
+    (values: SwapWarning) => {
+      'worklet';
+      swapWarning.modify(prev => ({ ...prev, ...values }));
+    },
+    [swapWarning]
+  );
 
+  const getWarning = useCallback(
```

Comment:
`getWarning` is called from a worklet, and its purpose is to call a different worklet. if feasible, it could be better to convert this to a worklet as well so all of this logic can remain on the UI thread, eliminating the need to switch threads twice (UI -> JS -> UI)

Reply:
could be not worth it since there's other non-worklet functionality in here

Reply:
can't use BigNumber math inside of a worklet so that's why it's structured this way. Agreed it's an anti-pattern but unsure how to fix it.